File: python/neetcode-150/003-two-sum.py
def twoSum(nums, target):
    diffMap = {}
    for i in range(len(nums)):
        diff  = target - nums[i]
        if nums[i] in diffMap:
            return [diffMap[nums[i]], i]
        diffMap[diff] = i
    return False

print(twoSum([2,7,11,15], 18))
        

File: 150-problems/003-two-sum.go
package problems

func TwoSum(nums []int, target int) []int {
    diffMap := make(map[int]int)

    for i, v := range nums {
		val, ok := diffMap[target - v]
		if ok {
			return []int{i, val}
		}
		diffMap[v] = i
	}

	return []int{-1, -1}
}